gammapy/astro/darkmatter/spectra.py

```diff
@@ -216,22 +214,28 @@ class DarkMatterAnnihilationSpectralModel(SpectralModel):
         interp="log",
     )
     scale._is_norm = True
+
+    jfactor = Parameter(
+        "jfactor",
+        1,
+        unit="GeV2 cm-5",
+        interp="log",
+    )
     tag = ["DarkMatterAnnihilationSpectralModel", "dm-annihilation"]
 
-    def __init__(self, mass, channel, scale=scale.quantity, jfactor=1, z=0, k=2):
+    def __init__(self, mass, channel, scale=scale.quantity, jfactor=jfactor, z=0, k=2):
         self.k = k
         self.z = z
         self.mass = u.Quantity(mass)
         self.channel = channel
-        self.jfactor = u.Quantity(jfactor)
         self.primary_flux = PrimaryFlux(mass, channel=self.channel)
-        super().__init__(scale=scale)
+        super().__init__(scale=scale, jfactor=jfactor)
```

Comment:
I agree on all four changes above, along the philosophy of making the `jfactor` a parameter